### PR TITLE
Add jobid checking to preprocess

### LINKED ISSUE #9: Incorrectly loading sdf data
When I load in the sdf file 0001.sdf I get the following in xarray for the `derived electron number density`: 
image

Whereas sdf_helper returns:
image

This is unexpected behaviour however at this moment in time I'm unsure how to fix the issue at hand. The setup used to generate this sdf file is hard to replicate so here's a link to it on Google Drive (annoyingly couldn't upload it here) https://drive.google.com/file/d/1LP6vEXk-FFGUHYhgLMUB48PN55hUjNJ4/view?usp=sharing

NOTE: If the file is opened directly using `xr.open_dataset(...)` it works fine but when using the following it fails:
```python
ds = xr.open_mfdataset(
    "/users/bmp535/scratch/epoch_runner/muons_campaign/run_0_1000000/run_0_10000/run_0_100/run_0/*.sdf",
    concat_dim="time",
    combine="nested",
    data_vars='minimal', 
    coords='minimal', 
    compat='override', 
    preprocess=lambda ds: ds.expand_dims(time=[ds.attrs["time"]])
)
```

### PR DESCRIPTION
**Pull request by JoelLucaAdams**
Add `jobid1` checking to the preprocess and separate out into new class that can be called from the code. Came about because of issue #9 

Did we want to make it the default option in `SDFEntrypoint` to avoid needing to call `from sdf_xarray import SDFPreprocess`?

Fixes #9

### CODE REVIEW

File: src/sdf_xarray/__init__.py [hunk 1]
```diff
@@ -228,3 +228,19 @@ def guess_can_open(self, filename_or_obj):
     description = "Use .sdf files in Xarray"
 
     url = "https://epochpic.github.io/documentation/visualising_output/python.html"
+
+
+class SDFPreprocess:
```
Comment (ZedThree) [inline src/sdf_xarray/__init__.py:233]: Just needs a docstring, then it's good to go!

**Review by ZedThree — APPROVED** [review]

File: src/sdf_xarray/__init__.py (hunk 1 above)
Comment (JoelLucaAdams) [reply src/sdf_xarray/__init__.py:233]: Done!

**Comment by ZedThree** [comment]
> Did we want to make it the default option in `SDFEntrypoint` to avoid needing to call `from sdf_xarray import SDFPreprocess`?

I don't think so, that would affect individual datasets. There doesn't seem to be a way to hook into `open_mfdataset` specifically.

**Comment by JoelLucaAdams** [comment]
Good point... pity that users need to define all those variables to get it working but for now I think it's fine

File: src/sdf_xarray/__init__.py (hunk 1 above)
Comment (JoelLucaAdams) [reply src/sdf_xarray/__init__.py:233]: Ah, my bad! fixed

**Pull request merged**